great_expectations/rule_based_profiler/config/base.py

```diff
@@ -408,32 +415,58 @@ def resolve_config_using_acceptable_arguments(
     ) -> "RuleBasedProfilerConfig":  # noqa: F821
         runtime_config: RuleBasedProfilerConfig = profiler.config
 
-        runtime_variables: Optional[
-            ParameterContainer
-        ] = profiler.reconcile_profiler_variables(variables)
-        runtime_variables_configs: dict = (
-            runtime_variables.to_dict()["parameter_nodes"]["variables"]["variables"]
-            or {}
+        runtime_variables: dict = profiler.reconcile_profiler_variables_as_dict(
+            variables=variables
         )
-
-        effective_rules: List["Rule"] = profiler.reconcile_profiler_rules(  # noqa: F821
-            rules=rules
+        variables_container: ParameterContainer = (
+            build_parameter_container_for_variables(variables_configs=runtime_variables)
         )
 
-        rule: "Rule"  # noqa: F821
-        runtime_rules: Dict[str, dict] = {
-            rule.name: rule.to_json_dict() for rule in effective_rules
-        }
+        effective_rules: Dict[
+            str, "Rule"  # noqa: F821
+        ] = profiler.reconcile_profiler_rules_as_dict(rules=rules)
+        runtime_rules: Dict[str, dict] = {}
+        for name, rule in effective_rules.items():
+            rule_with_substituted_vars: dict = (
+                RuleBasedProfilerConfig._interpolate_variables_in_config(
+                    rule=rule, variables_container=variables_container
+                )
+            )
+            runtime_rules[name] = rule_with_substituted_vars
 
         return cls(
             class_name=profiler.__class__.__name__,
             module_name=profiler.__class__.__module__,
             name=runtime_config.name,
             config_version=runtime_config.config_version,
-            variables=runtime_variables_configs,
+            variables=runtime_variables,
             rules=runtime_rules,
         )
 
+    @staticmethod
+    def _interpolate_variables_in_config(
+        rule: "Rule", variables_container: ParameterContainer
+    ) -> dict:
+        def _traverse_and_interpolate(node: Any) -> None:
+            if isinstance(node, dict):
+                for key, val in node.copy().items():
+                    if isinstance(val, str) and val.startswith(VARIABLES_KEY):
+                        node[key] = get_parameter_value_and_validate_return_type(
+                            domain=None,
+                            parameter_reference=val,
+                            variables=variables_container,
+                            parameters=None,
+                        )
+                    _traverse_and_interpolate(val)
+            elif isinstance(node, list):
+                for val in node:
+                    _traverse_and_interpolate(val)
```

Comment:
@cdkini positional vs. keyword arguments ... what is best?  I like keyword ones...  Sometimes, of course, positional is the only way.  Just a thought.

Reply:
don't think it's strictly necessary for 1-3 arg funcs but I agree in general